### PR TITLE
Allow password of type string as input

### PR DESCRIPTION
**Pull request by hyb175**
Allow password input of type string.
Added tests for allowing string input.
This is to address issue https://github.com/cjlarose/argon2-ffi/issues/1

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -49,6 +51,8 @@ function variant(hashRaw, hashEncoded, verify) {
 
     hash(...args) {
       const [password, salt, options, cb] = parseArgs(args);
+      let parsedPassword = password;
+      if (typeof password === 'string') { parsedPassword = new Buffer(password); }
```
Comment (cjlarose) [inline src/index.js]: What do you think about moving these two lines to `parseArgs`, so we don't have to repeat it in both `hashRaw` and `hash`?

**Comment by cjlarose** [comment]
Can you add tests? I think one test for each `hashRaw`, `hash`, and `verify` with a String password should be fine.

File: src/index.js (hunk 1 above)
Comment (hyb175) [inline src/index.js]: Sounds good

**Comment by cjlarose** [comment]
Nice work, Yingbai!

**Pull request merged**